safe/gui/tools/osm_downloader_dialog.py

```diff
@@ -174,15 +186,45 @@ def show_help(self):
         """Load the help text for the dialog."""
         show_context_help(self.help_context)
 
-    def update_extent(self):
-        """ Update extent value in GUI based from value in map."""
-        # Get the extent as [xmin, ymin, xmax, ymax]
-        extent = viewport_geo_array(self.iface.mapCanvas())
+    def update_extent(self, extent):
+        """ Update extent value in GUI based from an extent.
+
+        :param extent: a list in the form [xmin, ymin, xmax, ymax] where all
```

Comment:
A list .... (docstring parameter descriptions should start with a capital letter and be fully formed sentances).